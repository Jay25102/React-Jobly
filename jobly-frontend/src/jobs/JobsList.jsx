import React, { useState, useEffect } from "react";
import SearchForm from "../forms/SearchForm";
import JoblyApi from "../../api";
import JobCardList from "./JobCardList";
import LoadingSpinner from "../LoadingSpinner";

/**
 * Returns a list of jobs from the API. Uses the searchForm component
 * to filter jobs by name. Passes the list of jobs to JobCardList.
 */
function JobsList() {
    const [jobs, setJobs] = useState(null);
    console.debug("JobList");

    useEffect(function fetchAllJobsOnMount() {
        console.debug("JobList useEffect fetchAllJobsOnMount");
        search();
    }, []);
    
    async function search(title) {
        let jobs = await JoblyApi.getJobs(title);
        setJobs(jobs);
    }

    if (!jobs) return <LoadingSpinner/>;

    return (
        <div className="JobList col-md-8 offset-md-2">
            <SearchForm searchFor={search}/>
            {jobs.length 
                ? <JobCardList jobs={jobs}/>
                : <p className="lead">Sorry, no results were found!</p>
            }
        </div>
    )
}

export default JobsList;