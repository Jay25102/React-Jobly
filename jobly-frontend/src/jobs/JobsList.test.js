import React from "react";
import { render } from "@testing-library/react";
import JobsList from "./JobsList";

describe("JobsList", function() {
    test("it renders without crashing", function() {
        render(<JobsList/>);
    });
    test("it matches snapshot", function() {
        const { asFragment } = render(<JobsList/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
