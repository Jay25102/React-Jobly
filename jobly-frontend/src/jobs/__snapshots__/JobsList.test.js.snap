// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsList it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    loading...
  </div>
</DocumentFragment>
`;
